lms/djangoapps/discussion_api/api.py

```diff
@@ -583,7 +595,9 @@ def create_comment(request, comment_data):
         raise ValidationError({"thread_id": ["This field is required."]})
     try:
         cc_thread, context = _get_thread_and_context(request, thread_id)
-    except Http404:
+    except CourseNotFoundError:
+        raise ValidationError({"course_id": ["Invalid value."]})
+    except ThreadNotFoundError:
         raise ValidationError({"thread_id": ["Invalid value."]})
```

Comment:
In fact, the `ValidationError` in this file should be reserved for invalid argument values.  `CourseNotFoundError` and `ThreadNotFoundError`, however, are for exceptions when the object does not exist.  And those 2 errors can be raised/surfaced through the `create_comment` method.